doc/tutorials/core/univ_intrin/univ_intrin.markdown

```diff
@@ -12,20 +12,177 @@ Universal Intrinsics {#tutorial_univ_intrin}
+* *512 bit* registers are supported on x86(AVX512)
+
+@note To know more about what Instruction Sets are supported by your CPU, look [here]()
+
+
+We will now introduce the available structures and functions:
+* Register structures
+* Load and store
+* Mathematical Operations
+* Type conversions
+* Reduce and Mask
+
+### Register Structures
+
+The Universal Intrinsics set implements every register as a structure based on the particular SIMD register.
+All types contain the **nlanes** enumeration which gives the exact number of values that the type can hold. This eliminates the need to hardcode the number of values during implementations.
+
+@note Each register structure is under the *cv* namespace.
+
+There are **two types** of registers:
+
+* **Variable sized registers**: These structures do not have a fixed size and their exact bit length is deduced during compilation, based on the available SIMD capabilities. Consequently, the value of the **nlanes** enum is determined in compile time.
+    Each structure follows the following convention
+
+    ```
+    v_[type of value][size of each value in bits]
+    ```
+
+    For instance, **v_uint8 holds 8-bit unsigned integers** and **v_float32 holds 32-bit floating point values**. We then declare a register like we would declare any object in C++
+
+    <br>
+
+    Based on the available SIMD architecture, a particular register will hold different number of values.
+    For example: If your computer supports a maximum of 256bit registers,
+    * @ref cv::v_uint8 will hold 32 (*= 256 / 8*) 8-bit unsigned integers
+    * @ref cv::v_float64 will hold 4 (*= 256 / 64*) 64-bit floats (doubles)
+
+    <br>
+
+    Available data type and sizes:
+    |Type|Size in bits|
+    |-:|:-|
+    |uint| 8, 16, 32, 64|
+    |int | 8, 16, 32, 64|
+    |float | 32, 64|
+<br>
+
+* **Constant sized registers**: These structures have a fixed bit size and hold a constant number of values. We need to know what SIMD architecture is supported by the system and select compatible registers. Use these only if exact bit length is necessary.
+    <br>
+
+    Each structure follows the convention:
+
+    ```
+    v_[type of value][size of each value in bits]x[number of values]
+    ```
+    Suppose we want to store
+    * 32-bit(*size in bits*) signed integers in a **128 bit register**. Since the register size is already known, we can find out the *number of data points in register* (*128/32 = 4*): @ref v_int32x8
+
+        ```
+        v_int32x8 name_of_register // holds 8 32-bit signed integers.
+        ```
+    * 64-bit floats in 512 bit register: @ref v_float64x8
+
+        ```
+        v_float64x8 name_of_register // holds 8 64 bit floating point integers
+        ```
+
+@note Every 256-bit and 512-bit structure is a typedef of the v_reg<_Tp, n> class with the corresponding data type (_Tp) and the number of values (n). For using 256-bit(512-bit) registers, check the CV_SIMD256 (CV_SIMD512) preprocessor definition.
```

Comment:
Actually structures representing registers aren't based on v_reg for real architecture dependent implementations. v_reg is used to emulate hardware registers in scalar-based software implementation of intrinsics(which could be used if no hardware SIMD is available at all or for specific debugging cases)

Reply:
Could you elaborate on what this means? v_reg structures may represent both wide registers, if supported, and primitive registers in case of scaler implementations, right?

Reply:
'v_reg' structures used only for software implementation of Universal Intrinsics(or Wide Universal Intrinsics depending on configuration) that compute result using scalar code instead of actual SIMD CPU operations. This implementation could be used as a reference or for various specific configurations with special requirements. However all existing implementations for particular hardware map structures to physical registers in accordance with architecture/compiler guidance.